File: Classes/SYSC4005-Simulation/src/Buffer.py
import queue

from Component import Component

class Buffer:
    def __init__(self, name:str, maxsize=2) -> None:
        self.name = name
        self.buffer = queue.Queue(maxsize=maxsize)
    
    def __len__(self):
        return self.buffer.qsize()
    
    def put(self, component: Component):
        self.buffer.put(component)
        
    def get(self):
        return self.buffer.get()
    
    def get_name(self):
        return self.name

File: Classes/SYSC4005-Simulation/SYSC4005Project/src/main/java/org/example/Buffer.java
package org.example;

import java.util.ArrayList;

public class Buffer {
    public String name;
    public ArrayList<String> arrayList;
    int componentsIn, componentsOut;
    public Buffer(String name){
        this.name = name;
        this.arrayList = new ArrayList<String>();
        componentsIn = 0;
        componentsOut = 0;
    }

    public boolean isEmpty(){
        return arrayList.isEmpty();
    }

    public boolean isFull(){
        return (arrayList.size() >= 2);
    }

    public int getSize(){
        return arrayList.size();
    }

    public void addComponent(String component){
        arrayList.add(component);
        componentsIn++;
    }
    public void removeComponent(){
        arrayList.remove(0);
        componentsOut++;
    }

    public String getName(){
        return this.name;
    }

    public void printBufferInfo(){
        System.out.println("Buffer " + name + " had " + componentsIn + " components in and " + componentsOut +" components out.");
    }

}
